lit_parrot/speed_monitor.py

```diff
@@ -212,49 +215,32 @@ def on_train_batch_end(
                     "throughput/device/samples_per_sec": dev_samples_per_sec,
                 }
             )
-
             # Assumes no padding.
-            if max_seq_length is not None:
-                # Only applicable to seq data / models
+            if lengths is not None:
+                elapsed_lengths = int(self.history_lengths[-1]) - int(self.history_lengths[0])
                 metrics.update(
                     {
-                        "throughput/tokens_per_sec": samples_per_sec * max_seq_length,
-                        "throughput/device/tokens_per_sec": dev_samples_per_sec * max_seq_length,
+                        "throughput/tokens_per_sec": samples_per_sec * elapsed_lengths,
+                        "throughput/device/tokens_per_sec": dev_samples_per_sec * elapsed_lengths,
```

Comment:
This should be a more accurate (lower) estimate of the throughput in the cases where the sequence lengths are not equal to the model's block size.